substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfAbbrevSectionImpl.java

```diff
@@ -983,6 +987,10 @@ private int writeArrayUnitAbbrev(@SuppressWarnings("unused") DebugContext contex
         pos = writeAttrForm(DwarfDebugInfo.DW_FORM_data1, buffer, pos);
         pos = writeAttrType(DwarfDebugInfo.DW_AT_name, buffer, pos);
         pos = writeAttrForm(DwarfDebugInfo.DW_FORM_strp, buffer, pos);
+        pos = writeAttrType(DwarfDebugInfo.DW_AT_comp_dir, buffer, pos);
```

Comment:
Same thing, i think you need to write the corresponding values in `com.oracle.objectfile.elf.dwarf.DwarfInfoSectionImpl#writeArrayTypeUnit`